src/foam/support/view/TicketDetailView.js

```diff
@@ -130,12 +254,94 @@ foam.CLASS({
     }
   ],
   actions: [
+    {
+      name: 'submitTicket',
+      label: '',
+      code: function(){
+        if(this.viewData['variant']==false && this.messages=="" && this.data.requestorEmail!=""){
+        x = this.pop3;
+        var messageId=x.sendEmail(this.data.requestorEmail,this.data.subject,this.viewData['message']);
+        this.ticketDAO.find(this.data.id).then(function(a){
```

Comment:
We don't need this ticketDAO find.
We'll do this.data.emailId = messageId;
then:
this.ticketDAO.put(this.data)
this.stack.push...etc...